NGCHM/WebContent/javascript/MatrixManager.js

```diff
@@ -1726,14 +1707,33 @@ MMGR.HeatMapData = function(heatMapName, level, jsonData, datalayers, lowerLevel
     };
 
     /**********************************************************************************
-     * FUNCTION - zipAppDownload: This function calls the Matrix Manager to initiate
-     * the download of the NG-CHM File Viewer application.
+     * FUNCTION - zipAppDownload: The user clicked on the "Download Viewer" button.
+     * Hide the button and initiate download of the NG-CHM Viewer application.
      **********************************************************************************/
     MMGR.zipAppDownload = zipAppDownload;
     function zipAppDownload () {
-	    var dlButton = document.getElementById('msgBoxBtnImg_1');
-	    dlButton.style.display = 'none';
-	    downloadFileApplication();
+	const dlButton = document.getElementById('msgBoxBtnImg_1');
+	dlButton.style.display = 'none';
+	downloadFileApplication();
+    }
+
+    // Initiate download of NGCHM File Viewer application zip
+    function downloadFileApplication () {
+	if (typeof NgChm.galaxy !== 'undefined') {
+	    // Current viewer is embedded within Galaxy.
+	    // FIXME: BMB: Use a better way to determine Galaxy embedding.
+	    window.open("/plugins/visualizations/mda_heatmap_viz/static/ngChmApp.zip");
+	} else if (MMGR.getHeatMap().source() === MMGR.FILE_SOURCE) {
+	    // Heat map came from a disk file, not from a server.
+	    // (This does not mean the viewer is not from a server, so this could be
+	    // refined further for that case i.e. the "api" condition might be more appropriate)
+	    // Use full URL, which must be complete!
+	    callServlet("POST", COMPAT.viewerAppUrl, false);
+	} else {
+	    // Heat map came from a server.
+	    // Use server "api" + special endpoint name
+	    callServlet("POST", CFG.api + "ZipAppDownload", false);
```

Comment:
Maybe trivial, but why is this "POST"?  GET works (just now, from browser) and seems more appropriate.  Also true with line 1731.

Reply:
It used to be POST so I just kept that. I can change it.